lib/ansible/modules/system/mount.py

```diff
@@ -746,6 +752,18 @@ def main():
                 res, msg = mount(module, args)
 
         if res:
+            # Not restoring fstab after a failed mount was reported as bug #59183
+            try:
+                write_fstab(module, backup_lines, args['fstab'])
+            except Exception:
+                pass
+
+            try:
+                # Not exactly the reverse of os.makedirs(name), but consistent with state=absent
+                os.rmdir(name)
+            except Exception:
```

Comment:
What if name is /mnt?

Reply:
Will either fix or remove this logic. Thanks!

Reply:
Well, the new code is consistent with the existing logic regarding `state == 'absent'`: it would remove `/mnt`, too, and there is no possibility to implement the "if the module created it" test. So I see no sane solution other than never removing the mount point. If you have a better idea, you are welcome to express it.

Reply:
Sorry, this is my confusion between 'absent' and 'unmounted'. Will implement the "remove dirs that we created" logic.

Reply:
Fixed the objection, please look again.